pkg/image/validation.go

```diff
@@ -159,5 +165,95 @@ func validateSuma(os *OperatingSystem) error {
 	if os.Suma.ActivationKey == "" {
 		return fmt.Errorf("no activation key defined")
 	}
+
+	return nil
+}
+
+func validateHauler(definition *Definition) error {
+	if checkIfHaulerDefined(definition) {
+		return nil
+	}
+	err := validateContainerImages(definition.Hauler.ContainerImages)
+	if err != nil {
+		return fmt.Errorf("error validating container images: %w", err)
+	}
+	err = validateHelmCharts(definition.Hauler.HelmCharts)
+	if err != nil {
+		return fmt.Errorf("error validating helm charts: %w", err)
+	}
+	err = validateFiles(definition.Hauler.Files)
+	if err != nil {
+		return fmt.Errorf("error validating files: %w", err)
+	}
+
+	return nil
+}
+
+func checkIfHaulerDefined(definition *Definition) bool {
+	return len(definition.Hauler.Files) == 0 &&
+		len(definition.Hauler.HelmCharts) == 0 &&
+		len(definition.Hauler.ContainerImages) == 0
+}
+
+func validateContainerImages(containerImages []ContainerImage) error {
+	seenContainerImages := make(map[string]bool)
+
+	for _, image := range containerImages {
+		if image.Name == "" {
+			return fmt.Errorf("no image name defined")
+		}
+
+		if seenContainerImages[image.Name] {
+			return fmt.Errorf("duplicate container image found: '%s'", image.Name)
+		}
+		seenContainerImages[image.Name] = true
+	}
+
+	return nil
+}
+
+func validateHelmCharts(charts []HelmChart) error {
+	seenCharts := make(map[string]bool)
+
+	for _, chart := range charts {
+		if chart.Name == "" {
+			return fmt.Errorf("no chart name defined")
+		}
+		if chart.RepoURL == "" {
+			return fmt.Errorf("no chart repository URL defined")
+		}
+		if chart.Version == "" {
+			return fmt.Errorf("no chart version defined")
+		}
+		if !strings.HasPrefix(chart.RepoURL, "http") {
+			return fmt.Errorf("invalid chart respository url, does not start with 'http://' or 'https://'")
+		}
+
+		if seenCharts[chart.Name] {
+			return fmt.Errorf("duplicate chart found: '%s'", chart.Name)
+		}
+		seenCharts[chart.Name] = true
+	}
+
+	return nil
+}
+
+func validateFiles(files []File) error {
+	seenFiles := make(map[string]bool)
+
+	for _, file := range files {
+		if file.Name == "" {
+			return fmt.Errorf("no file name defined")
+		}
+		if file.Path == "" {
+			return fmt.Errorf("no file path defined")
```

Comment:
Can we include the name of the file in this wrapper?

Reply:
resolving as a result of the removal of hauler file handling